libraries/Microsoft.Bot.Connector/OAuthClientConfig.cs

```diff
@@ -45,92 +47,88 @@ public static async Task SendEmulateOAuthCardsAsync(OAuthClient client, bool emu
-            string requestContent = null;
-
-            // Set Credentials
-            if (client.Credentials != null)
+            using (var httpRequest = new HttpRequestMessage())
             {
-                cancellationToken.ThrowIfCancellationRequested();
-                await client.Credentials.ProcessHttpRequestAsync(httpRequest, cancellationToken).ConfigureAwait(false);
-            }
+                httpRequest.Method = new HttpMethod("POST");
+                httpRequest.RequestUri = new Uri(url);
 
-            // Send Request
-            if (shouldTrace)
-            {
-                ServiceClientTracing.SendRequest(invocationId, httpRequest);
-            }
+                var cancellationToken = CancellationToken.None;
 
-            cancellationToken.ThrowIfCancellationRequested();
-            httpResponse = await client.HttpClient.SendAsync(httpRequest, cancellationToken).ConfigureAwait(false);
-            if (shouldTrace)
-            {
-                ServiceClientTracing.ReceiveResponse(invocationId, httpResponse);
-            }
+                // Serialize Request
+                string requestContent = null;
 
-            HttpStatusCode statusCode = httpResponse.StatusCode;
-            cancellationToken.ThrowIfCancellationRequested();
-            string responseContent = null;
-            if (statusCode != HttpStatusCode.OK && statusCode != HttpStatusCode.NotFound)
-            {
-                var ex = new ErrorResponseException(string.Format("Operation returned an invalid status code '{0}'", statusCode));
-                try
+                // Set Credentials
+                if (client.Credentials != null)
                 {
-                    responseContent = await httpResponse.Content.ReadAsStringAsync().ConfigureAwait(false);
-                    ErrorResponse errorBody = Rest.Serialization.SafeJsonConvert.DeserializeObject<ErrorResponse>(responseContent, client.DeserializationSettings);
-                    if (errorBody != null)
-                    {
-                        ex.Body = errorBody;
-                    }
+                    cancellationToken.ThrowIfCancellationRequested();
+                    await client.Credentials.ProcessHttpRequestAsync(httpRequest, cancellationToken).ConfigureAwait(false);
                 }
-                catch (JsonException)
+
+                // Send Request
+                if (shouldTrace)
                 {
-                    // Ignore the exception
+                    ServiceClientTracing.SendRequest(invocationId, httpRequest);
                 }
 
-                ex.Request = new HttpRequestMessageWrapper(httpRequest, requestContent);
-                ex.Response = new HttpResponseMessageWrapper(httpResponse, responseContent);
+                cancellationToken.ThrowIfCancellationRequested();
+                var httpResponse = await client.HttpClient.SendAsync(httpRequest, cancellationToken).ConfigureAwait(false);
```

Comment:
`HttpResponseMessage` implements IDisposable. Might be a good opportunity to wrap this in a using.

Reply:
Done